### ISSUE TITLE
v1.6.1 centraldashboard error logging prints [object Object]

### ISSUE THREAD

**Issue by adobemomo** [kubeflow/kubeflow#7232]
/kind bug

**What steps did you take and what happened:**
When visiting centraldashboard, namespaces can not be listed. Then I checked the console, it's showing env-info GET 503.
<img width="931" alt="image" src="https://github.com/kubeflow/kubeflow/assets/39028148/597eb5f9-7042-4bb2-9af4-8ba1d430ab1b">

Then I checked the central dashboard log, it's showing `Unable to get environment info: [object Object]`. The error message is not printed.

**What did you expect to happen:**
The error message detail should be printed rather than [object Object]

**Anything else you would like to add:**

This problem lies in this piece of code:
`centraldashboard-v1.6.1/app/api_workgroup.ts`
```
        .get('/env-info', async (req: Request, res: Response) => {
            try {
                if (req.user.hasAuth) {
                    return res.json(await this.getProfileAwareEnv(req.user));
                }
                res.json(await this.getBasicEnvironment(req.user));
            } catch (err) {
                const code = (err.response && err.response.statusCode) || 400;
                const error = err.body || 'Unexpected error getting environment info';
                console.log(`Unable to get environment info: ${error}${err.stack?'\n'+err.stack:''}`);
                apiError({res, code, error});
            }
        })
```

After I change it to the following code, it prints as expected:
```
        .get('/env-info', async (req: Request, res: Response) => {
            try {
                if (req.user.hasAuth) {
                    return res.json(await this.getProfileAwareEnv(req.user));
                }
                res.json(await this.getBasicEnvironment(req.user));
            } catch (err) {
                const code = (err.response && err.response.statusCode) || 400;
                const error = err.message || err.body || 'Unexpected error getting environment info';
                console.log(`Unable to get environment info: ${JSON.stringify(err, null, 2)}${err.stack?'\n'+err.stack:''}`);
                apiError({res, code, error});
            }
        })
```

**Environment:**

- Kubeflow version: (version number can be found at the bottom left corner of the Kubeflow dashboard): 1.6.1
- kfctl version: (use `kfctl version`): N/A
- Kubernetes platform: Azure Kubernetes Service
- Kubernetes version: (use `kubectl version`): Client: v1.23.0, Server: v1.24.10
- OS (e.g. from `/etc/os-release`): MacOS 13.5

**Comment by juliusvonkohout** [comment]
Please update to Kubeflow 1.7/1.8

**Comment by andreyvelich** [comment]
Feel free to re-open if it is not resolved in Kubeflow >= 1.7
